src/services/AnalyticsService.js

```diff
@@ -0,0 +1,152 @@
+  trackFeedback(feedbackComponent, isFakeUser) {
+    if (isFakeUser) return;
+
+    let aggResponses = [];
+    let volunteerScore = 0;
+
+    // creates an array of an agreggate of all responses and subresponses
+    for (var response in feedbackComponent.userResponse) {
+      if (typeof feedbackComponent.userResponse[response] === "object") {
+        for (var subresponse in feedbackComponent.userResponse[response]) {
+          aggResponses.push(
+            feedbackComponent.userResponse[response][subresponse]
+          );
+        }
+      } else {
+        aggResponses.push(feedbackComponent.userResponse[response]);
+      }
+    }
+    // adds to volunteer score
+    if (feedbackComponent.userType === "student") {
+      volunteerScore = aggResponses.reduce(function(acc, val) {
+        return acc + val;
+      }, 0);
+    }
+    // sends information
+    window.analytics.track("feedback", {
+      "session id": feedbackComponent.sessionId,
+      user: feedbackComponent.userType,
+      "student id": feedbackComponent.studentId,
+      "volunteer id": feedbackComponent.volunteerId,
+      "volunteer score": volunteerScore
+      // can get answers to specific response using aggResponses
+    });
+  },
+
+  // tracks when a help session has ended
+  trackSessionEnded(headerComponent, currentSession, isFakeUser) {
+    if (isFakeUser) return;
+
+    // calculating time-related session info (session length, wait time, etc.)
+    let volunteerSessionLength = null;
+    let waitTime = null;
+    let volunteerShowed = null;
+    const timeSessionEnded = new Date();
+    const timeCreatedAt = new Date(currentSession.createdAt);
+
+    if (currentSession.volunteerJoinedAt) {
+      volunteerShowed = new Date(currentSession.volunteerJoinedAt);
+      waitTime = ((volunteerShowed - timeCreatedAt) / 60000).toFixed(2);
+      volunteerSessionLength = (
+        (timeSessionEnded - volunteerShowed) /
+        60000
+      ).toFixed(2);
+    }
+
+    const sessionLength = ((timeSessionEnded - timeCreatedAt) / 60000).toFixed(
+      2
+    );
+
+    // getting if messages were exchanged
+    let studentMessages = 0;
+    let volunteerMessages = 0;
+    let successfulSession = false;
+
+    // loops through messages, and counts how many were by student vs. volunteer
+    for (var message in currentSession.messages) {
+      const isStudentMessage =
+        currentSession.messages[message].user === currentSession.student._id;
+      const isVolunteerMessage =
+        currentSession.volunteer &&
+        currentSession.messages[message].user === currentSession.volunteer._id;
+
+      if (isStudentMessage) {
+        studentMessages++;
+      } else if (isVolunteerMessage) {
+        volunteerMessages++;
+      }
+    }
+
+    // current criteria for a successful session
+    if (studentMessages > 0 && volunteerMessages > 0) {
+      successfulSession = true;
+    }
+
+    window.analytics.track("session ended", {
```

Comment:
Hmmm, on staging, I'm not seeing the "session end" event get sent. The "session started" event shows up just fine in the Segment debugger, along with "logged in" and "feedback." Could you take a look at that?

Reply:
Yup, fixed it!